cinema/serializers.py

```diff
@@ -18,7 +18,50 @@
             "description", instance.description
         )
         instance.duration = validated_data.get("duration", instance.duration)
+        instance.save()
+        return instance
+
+
+class GenreSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(max_length=255)
+
+    def create(self, validated_data):
+        return Genre.objects.create(**validated_data)
 
+    def update(self, instance, validated_data):
+        instance.name = validated_data.get("name", instance.name)
         instance.save()
+        return instance
+
 
+class ActorSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    first_name = serializers.CharField(max_length=255)
+    last_name = serializers.CharField(max_length=255)
+
+    def create(self, validated_data):
+        return Actor.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        instance.first_name = validated_data.get("first_name", instance.first_name)
+        instance.last_name = validated_data.get("last_name", instance.last_name)
+        instance.save()
+        return instance
+
+
+class CinemaSerializer(serializers.Serializer):
```

Comment:
Consider using `serializers.ModelSerializer` instead of `serializers.Serializer` for the `CinemaSerializer`. This will automatically generate fields based on the model and reduce boilerplate code.